File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    return dfs(numbers, 0, 0, target);
}

function dfs(numbers, index, sum, target){
    if(numbers.length===index&&target===sum)
        return 1;
    else if(numbers.length===index)
        return 0;
    
    return dfs(numbers, index+1, sum + numbers[index], target) + dfs(numbers, index+1, sum - numbers[index], target)
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    def dfs(numbers, index, sum, target):
        if index==len(numbers) and sum==target:
            return 1
        elif index==len(numbers):
            return 0
        return dfs(numbers, index+1, sum+numbers[index], target) + dfs(numbers, index+1, sum-numbers[index], target)
    
    return dfs(numbers, 0, 0, target)